app/org/sagebionetworks/bridge/services/email/WithdrawConsentEmailProvider.java

```diff
@@ -0,0 +1,89 @@
+import org.sagebionetworks.bridge.models.accounts.User;
+import org.sagebionetworks.bridge.models.accounts.Withdrawal;
+import org.sagebionetworks.bridge.models.studies.Study;
+
+public class WithdrawConsentEmailProvider implements MimeTypeEmailProvider {
+    
+    private static final DateTimeFormatter FORMATTER = DateTimeFormat.forPattern("MMMM d, yyyy");
+    private static final String CONSENT_EMAIL_SUBJECT = "Notification of consent withdrawal for %s";
+    private static final String SUB_TYPE_HTML = "html";
+
+    private Study study;
+    private String externalId;
+    private User user;
+    private Withdrawal withdrawal;
+    
+    public WithdrawConsentEmailProvider(Study study, String externalId, User user, Withdrawal withdrawal) {
+        this.study = study;
+        this.externalId = externalId;
+        this.user = user;
+        this.withdrawal = withdrawal;
+    }
+    
+    @Override
+    public MimeTypeEmail getEmail(String defaultSender) throws MessagingException {
+        MimeTypeEmailBuilder builder = new MimeTypeEmailBuilder();
+
+        String subject = String.format(CONSENT_EMAIL_SUBJECT, study.getName());
+        builder.withSubject(subject);
+
+        final String sendFromEmail = isNotBlank(study.getSupportEmail()) ? 
+                String.format("%s <%s>", study.getName(), study.getSupportEmail()) : defaultSender;
+        builder.withSender(sendFromEmail);
+
+        Set<String> emailAddresses = commaListToSet(study.getConsentNotificationEmail());
+        for (String email : emailAddresses) {
+            builder.withRecipient(email.trim());
+        }
+
+        String content = String.format("<p>User %s withdrew from the study on %s. </p>", 
+                getUserLabel(), FORMATTER.print(withdrawal.getWithdrewOn()));
+        content += "<p>Reason:</p>";
+        if (StringUtils.isBlank(withdrawal.getReason())) {
+            content += "<p><i>No reason given.</i></p>";
+        } else {
+            content += "<p>"+withdrawal.getReason()+"</p>";
+        }
+        
+        // Consent agreement as message body in HTML
+        final MimeBodyPart bodyPart = new MimeBodyPart();
+        bodyPart.setText(content, StandardCharsets.UTF_8.name(), SUB_TYPE_HTML);
+        builder.withMessageParts(bodyPart);
+
+        return builder.build();
+    }
+    
+    private String getUserLabel() {
+        String label = String.format("%s %s &lt;%s&gt;", nullSafe(user.getFirstName()), nullSafe(user.getLastName()), user.getEmail());
+        if (externalId != null) {
+            label += " (external ID: " + externalId + ") ";
+        }
+        return label;
+    }
+
+    private String nullSafe(String value) {
+        return (value == null) ? "" : value;
+    }
+    
+    private Set<String> commaListToSet(String commaList) {
+        if (isNotBlank(commaList)) {
+            return commaDelimitedListToSet(commaList);
+        }
+        return Collections.emptySet();
+    }
```

Comment:
This is dupe code of something in ConsentEmailProvider? Can we move it to a shared util?